### DISCUSSION TITLE
Different overlays for different stream profiles

### DISCUSSION THREAD

**Discussion by Pedro-Leitek** [AxisCommunications/discussions#101] [VAPIX] [answered]
Hi

I need the following: When I use the api for retrieving a jpeg (jpg/image.cgi) I need the image to have no overlay. But the rtsp should have that.
Is there any way that I can disable the overlay when using jpg/image.cgi?

Thanks

**Comment by haichaoji123** [comment, marked as answer]
you can control to add overlay to some resolution.

here is the control callback:
settings.select_callback=select_callback;//setup here to only display overlay on selected resolution


/**
 * Callback that can be used to select which streams to render overlays
 * to. Note that YCBCR streams are always skipped since these are used
 * for analytics.
 *
 * @param camera The id of the camera that the stream is for.
 * @param width The width of the stream For further details see the comments
 *              for the stream_data struct.
 * @param height The height of the stream For further details see the comments
 *               for the stream_data struct.
 * @param rotation The rotation of the stream.
 * @param is_mirrored TRUE if mirroring is enabled for the stream.
 * @return Returns TRUE if overlays should be rendered for the stream.
 */
static gboolean
select_callback(gint camera, gint width,
			gint height, gint rotation,
			gboolean is_mirrored,
			enum axoverlay_stream_type type)
{
  if(height==720 && width==1280)
	return TRUE;
  else
	return FALSE;
}
